.github/workflows/pr_basic_checks.yml

```diff
@@ -0,0 +1,243 @@
+# This workflow performs the checks like license check,
+# doxygen, unit tests etc.
+
+on:
+  pull_request:
+
+jobs:
+  basic-checks:
```

Comment:
instead of keep this as blocks of `basic-checks` (Which is duplicated name with top level name)
can we break this into several jobs, so they can run in parallel, and more elaborated on what is the test about?
- License Check
- Include Check
- Coding Styles Check (Astyle + UTF)
- Docs Check (Doxygen + Spelling)

Reply:
I think it makes sense.